tool/gravity/cli/config.go

```diff
@@ -865,12 +868,42 @@ type agentConfig struct {
 	cloudProvider string
 }
 
+func retryUpdateJoinConfigFromCloudMetadata(ctx context.Context, config *autojoinConfig) error {
+	// Security: Insecure is OK here because we're only testing reachability to the cluster
+	client := httplib.GetClient(true,
+		httplib.WithTimeout(5*time.Second),
+		httplib.WithInsecure())
```

Comment:
nit: this will add `WithInsecure` twice (boolean argument also means `insecure`). Would be nice to get rid of the boolean argument in the first place (although this is probably out of this scope) as redundant and better make do with functional options.